charts/kong/templates/_helpers.tpl

```diff
@@ -14,7 +14,7 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 
 {{- define "kong.fullname" -}}
 {{- $name := default .Chart.Name .Values.nameOverride -}}
-{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- default (printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-") .Values.fullnameOverride -}}
```

Comment:
One small comment on this though: we might want to consider failing on templates rendering when `.Values.fullnameOverride` is longer than 63 characters since that might cause problems.

Reply:
Not sure if I understood but my proposal was not to truncate the full override but to [`fail`](https://helm.sh/docs/chart_template_guide/function_list/#fail) and "return an error" (to the extent possible with helm).